feat: demo flow

matteojug opened the pull request:
Some scaffolding (from sBTC) and initial implementation.

There are basically no tests, but can be tested with sBTC devenv (via a manual deposit, I'll check with pox payments later):
 - Usual `make devenv-up`, wait for nakamoto and `./signers.sh demo` for donation+sanity check
 - Get signers aggregate key with `cargo run -p signer --bin demo-cli info`
 - Edit `signer/src/bin/demo_cli.rs`, `exec_deposit` to return after `send_raw_transaction` but before `create_deposit`
 
Now, in no particular order:
 - Start spox: `cargo run -- -c src/config/default.toml --signers-xonly <signers xonly pubkey from info above>`
 - Create a deposit (without notifying emily): `cargo run -p signer --bin demo-cli deposit --amount 123456`

This will look for deposits made to the signers pubkey with the devenv default values. Once the tx is confirmed it should appear on Emily, assuming it didn't expire in the meantime, and be processed by the signers, assuming the amount is not too low to be ignored.

DRAFT: basically looking for general comments, then I can split this in different PRs:
 - https://github.com/BitcoinL2-Labs/sPoX/pull/2
 - https://github.com/BitcoinL2-Labs/sPoX/pull/3
 - https://github.com/BitcoinL2-Labs/sPoX/pull/4
 - https://github.com/BitcoinL2-Labs/sPoX/pull/5
 - https://github.com/BitcoinL2-Labs/sPoX/pull/6
 - https://github.com/BitcoinL2-Labs/sPoX/pull/7

djordon commented on src/deposit_monitor.rs:
```diff
@@ -0,0 +1,123 @@
+        &self,
+        chain_tip: &BlockRef,
+    ) -> Result<Vec<CreateDepositRequestBody>, Error> {
+        // TODO: cache this
+        let monitored_addrs = self.monitored.keys().cloned().collect::<Vec<_>>();
```
I don't think that we need to cache this.

djordon commented on src/deposit_monitor.rs:
```diff
@@ -0,0 +1,123 @@
+
+        let create_deposits = utxos
+            .iter()
+            .flat_map(|utxo| {
+                self.get_deposit_from_utxo(utxo, chain_tip)
```
Hurray for sync code, makes this so easy.

djordon reviewed (commented):
Just ran this on devenv and it worked flawlessly 🚀 🚀 🚀 

Edit: As for direction, this is great. It's basically what I had in mind for the implementation.

matteojug commented:
Closing this since the breakout PRs are up, the last one can be used to test the whole thingy (https://github.com/BitcoinL2-Labs/sPoX/pull/7). I should have addressed all the comments in the respective PR.
